packages/block-library/src/image/edit.js

```diff
@@ -245,6 +246,16 @@ class ImageEdit extends Component {
 		};
 	}
 
+	getFilename( url ) {
+		if ( url ) {
+			const fileName = url.toString().match( /.*\/(.+?)$/ );
```

Comment:
It looks like the regular expression doesn't take query string parameters into account, and I think it should strip them.

For example, the image for my gh avatar has some params in the url -> https://avatars2.githubusercontent.com/u/677833?s=60&v=4

I also think the toString might be unnecessary, I'd expect url to be a string or undefined given that's its attribute type.

Reply:
Thanks! Nice callout. My regex is horrible so I would love a review on my recent commit. I tested with a regular media library image and also the avatar string you provided. Seems to be working! 😬

Reply:
One change I'd make is to for the second group to be a non-capturing group, since we're not concerned with using its matched content:
```javascript
/.*\/(.+?)(?:\?.*)?$/
```

Reply:
Cool! Good call. Thanks!